architecture/core.rst

```diff
@@ -390,6 +390,186 @@ normal world supports scheduling of processes. Even on UP systems, supporting
 several trusted threads in optee_os helps normal world scheduler to be
 efficient.
 
+Core handlers for native interrupts
+===================================
+
+OP-TEE core offers means to implement native interrupt
+(see :ref:`native_foreign_irqs`) controller drivers and means for device
+drivers to configure and register handler functions for these interrupts.
+Interrupt handlers can be nested for when an interrupt controller
+exposes interrupts which signaling is multiplexed on an interrupt
+controlled by a parent interrupt controller.
+
+Interrupt controllers are represented by an instance of ``struct itr_chip``.
+An interrupt controller exposes a given number of interrupts, each identified
+by an index from 0 to N-1 where N is the total number of interrupts exposed
+by that controller. In the litterature, an interrupt index identifier
+is called interrupt number.
+
+The root interrupt controller currently implemented in OP-TEE core is a
+GIC instance for the Arm architecture.
+
+**Interrupt management API functions**
+
+Interrupt management resources are declared in header file
+kernel/interrupt.h_. Main API function are:
+
+    - ``interrupt_enable()`` and ``interrupt_disable()`` to respectively
+      enable and disable an interrupt.
+
+    - ``interrupt_mask()`` and ``interrupt_unmask()`` to respectively
+      mask/unmask an interrupt.
+
+    - ``interrupt_configure()`` to configure an interrupt detection mode
+      and priority.
+
+    - ``interrupt_add_handler()`` and ``interrupt_remove_handler()`` to
+      respectively register and unregister an interrupt handler function
+      for a given interrupt.
+
+When configuration switch ``CFG_DRIVERS_INTERRUPT`` is enabled, core
+offers a means to use Device Tree to describe interrupts topologyin the
+platform. When used, the platform must provide a secure devicetree blob
```

Comment:
There is a spelling error "topologyin", but I suggest rewriting the sentence slightly.
```suggestion
When the configuration switch ``CFG_DRIVERS_INTERRUPT`` is enabled,
the core provides a way to describe the platform's interrupt topology
using Device Tree.
```

Reply:
The "When used, ..." should be there. It's just the one sentence I had in mind.

Reply:
Thanks. I will also replace `CFG_DRIVERS_INTERRUPT` with `CFG_DT`. `CFG_DRIVERS_INTERRUPT` is useless.